src/nb_enum.cpp

```diff
@@ -150,6 +173,21 @@ bool enum_from_python(const std::type_info *tp, PyObject *o, int64_t *out, uint8
     if (!t)
         return false;
 
+    if ((t->flags & (uint32_t) enum_flags::flag_enum) != 0 && Py_TYPE(o) == t->type_py) {
+        auto pValue = PyObject_GetAttrString(o, "value");
+        if (pValue == nullptr) {
+            PyErr_Clear();
+            return false;
+        }
+        long long value = PyLong_AsLongLong(pValue);
```

Comment:
See above for the issue with signed 64 bit integers and all bits being set.

Reply:
It's not uncommon to have an enumeration like

```cpp
enum class Foo : uint64_t {
     A = 1 << 0,
     B = 1 << 1,
     // ...
     All = (uint64_t) -1
};
```
I'm worried that this won't pass the tests because it cannot be converted to a `long long`. In fact, I think it would be good to add an explicit testcase for a flag enum since an all-bits-1 enumerant is such a standard pattern there.

Reply:
Hi @hpkfft--I think that would be the case if we're doing arithmetic causing overflow with a type, but is that the case? AFAIK we're just casting `long long` / `unsigned long long` into `int64_t` and back.